onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -37,11 +37,78 @@ BasicBackend::BasicBackend(const ONNX_NAMESPACE::ModelProto& model_proto,
     : global_context_(global_context), subgraph_context_(subgraph_context) {
   std::string& hw_target = (global_context_.device_id != "") ? global_context_.device_id : global_context_.device_type;
   bool vpu_status = false;
-  bool import_blob_status = false;
   std::string model_blob_name;
   std::ifstream blob_path;
   std::string ov_compiled_blobs_dir = "";
 
+#if defined(OPENVINO_2021_4)
+  if(hw_target == "MYRIAD")
+    vpu_status = true;
+  const std::string compiled_blob_path = onnxruntime::GetEnvironmentVar("OV_BLOB_PATH");
+  if (vpu_status == true && openvino_ep::backend_utils::UseCompiledNetwork() && !compiled_blob_path.empty() &&
+      openvino_ep::BackendManager::GetGlobalContext().is_wholly_supported_graph) {
+    try {
+        LOGS_DEFAULT(INFO) << log_tag << "Importing the pre-compiled blob from the path set by the user";
+        exe_network_ = global_context_.ie_core.ImportNetwork(compiled_blob_path, hw_target, {});
+    } catch (Exception &e) {
+      ORT_THROW(log_tag + " Exception while Importing Network for graph: " + subgraph_context_.subgraph_name + ": " + e.what());
+    } catch(...) {
+      ORT_THROW(log_tag + " Exception while Importing Network for graph: " + subgraph_context_.subgraph_name);
+    }
+    LOGS_DEFAULT(INFO) << log_tag << "Succesfully Created an executable network from a previously exported network";
+  } else {
+  ie_cnn_network_ = CreateCNNNetwork(model_proto, global_context_, subgraph_context_, const_outputs_map_);
+  SetIODefs(model_proto, ie_cnn_network_, subgraph_context_.output_names, const_outputs_map_, global_context_.device_type);
+
+  if (const_outputs_map_.size() == subgraph_context_.output_names.size())
+    subgraph_context_.is_constant = true;
+
+  // Loading model to the plugin
+  if (subgraph_context_.is_constant) {
+    LOGS_DEFAULT(INFO) << log_tag << "The subgraph is a const. Directly moving to Infer stage.";
+    return;
+  }
+
+  std::map<std::string, std::string> config;
+#ifndef NDEBUG
+  if (openvino_ep::backend_utils::IsDebugEnabled()) {
+    config["PERF_COUNT"] = CONFIG_VALUE(YES);
+  }
+#endif
+  if (global_context_.device_type.find("MYRIAD") != std::string::npos) {
```

Comment:
Havent we checked earlier that it was myriad device... is it under the same loop and duplicated condition

Reply:
Line No: 78, where we are checking if it is Myriad device again is different case.

Here we are at the else condition for the flow #if defined(OPENVINO_2021_4), where the code is common for all the devices. so, here we are checking only if it is myriad, then add these additional configs else don't.